### PR TITLE
New ECS powers sdk supported

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
This sdk supports ECS instance to power on and power off.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
1. remove invalid sdk and related functions (power on: Start(), power off: Stop()).
2. new functions of batch power on and off.
```

### CODE REVIEW

File: openstack/ecs/v1/powers/requests.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	ShowdownType string       `json:"type,omitempty"`
+}
+
+type ServerInfo struct {
+	ID string `json:"id" required:"true"`
+}
+
+type PowerOptsBuilder interface {
+	ToPowerOnMap() (map[string]interface{}, error)
+	ToPowerOffMap() (map[string]interface{}, error)
```
Comment (ShiChangkuo) [inline openstack/ecs/v1/powers/requests.go:19]: can we use `ToPowerAction(action string) (map[string]interface{}, error)` ?
then we can do more actions into one interface

File: openstack/ecs/v1/powers/requests.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	if err != nil {
+		r.Err = err
+		return
+	}
+
+	_, r.Err = client.Post(actionURL(client), reqBody, &r.Body, &golangsdk.RequestOpts{OkCodes: []int{200}})
+	return
+}
+
+func PowerOff(client *golangsdk.ServiceClient, opts PowerOptsBuilder) (r cloudservers.JobResult) {
```
Comment (ShiChangkuo) [inline openstack/ecs/v1/powers/requests.go:41]: reboot action is missing

Comment (Lance52259) [reply openstack/ecs/v1/powers/requests.go]: thx, the function is supports power off/on and reboot now.

File: openstack/ecs/v1/powers/requests.go (hunk 1 above)
Comment (Lance52259) [reply openstack/ecs/v1/powers/requests.go]: more actions can supported with `ToPowerActionMap` into one interface result

File: openstack/ecs/v1/powers/requests.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+type ServerInfo struct {
+	ID string `json:"id" required:"true"`
+}
+
+type PowerOptsBuilder interface {
+	ToPowerActionMap(option string) (map[string]interface{}, error)
+}
+
+func (opts PowerOpts) ToPowerActionMap(option string) (map[string]interface{}, error) {
```
Comment (ShiChangkuo) [inline openstack/ecs/v1/powers/requests.go:21]: suggest to add more comments about the valid vaules of `option`

**Pull request merged**